### ISSUE TITLE
Support enums introduced in PHP 8.1

### ISSUE THREAD

**Issue by andrzejkupczyk** [tighten/ziggy#499]
#### Description

In Laravel, there is a way to resolve route binding to any object (not only Eloquent model), as long as the class implements the `UrlRoutable` contract. It can be for [example](https://gist.github.com/andrzejkupczyk/8436f9cc321d5d7bafd04a3f4b753720) an [enum](https://wiki.php.net/rfc/enumerations), although currently this is not supported.

As enum may not include `__construct`, Laravel is not able to resolve it from the container using the `app()` helper function in `Ziggy.php`:
```php
// Avoid booting this model if it doesn't override the default route key name
$bindings[$parameter->getName()] = $override ? app($model)->getRouteKeyName() : 'id';
```

#### Suggestion

IMO it would be great if different resolvers could be used (per parameter maybe? based on type or interfaces it implements?). Default resolver - for backward compatibility - would be still using the `app()` helper function. This could be configured in the Ziggy's configuration file.

#### Alternatives

The simplest solution would be to pass any useful data as parameters, for example:
```php
$bindings[$parameter->getName()] = $override
    ? app($model, ['reflection' => new ReflectionClass($model)])->getRouteKeyName()
    : 'id';
```

This way it would be possible to customize registration of a binding with the container.

**Comment by bakerkretzmar** [comment]
Support the recent Enum addition would be awesome but I don't think I follow your explanation of how that would work. What's the difference between your two code examples? The second one still uses the `app()` helper, it just passes one additional parameter?

**Comment by andrzejkupczyk** [comment]
Hey Jacob,

the first one is an excerpt from current implementation, which is causing mentioned problem with enums. This is not an example 🙂

Now, as regards the second one (which is just a simpler alternative to the preferred solution, i.e. dedicated resolvers):

> (…) it just passes one additional parameter?

Correct! This _minor_ change could do the trick. As I wrote, it could provide any useful detail about the route parameter (not necessarily a `ClassReflection` 😉)

Imagine your route definition is like this (where `ReviewSection` is an enum and not a Eloquent model):
```php
// e.g. web.php
Route::get('reviews/:id/sections/{section}', function (ReviewSection $section) {
    return $section;
});
```

Having this additional parameter(s), containing for example value of the `{section}` parameter, would allow you to properly configure binding with the service container:
```php
// e.g. AppServiceProvider
$this->app->bind(ReviewSection::class, function ($app, $parameters) {
    return ReviewSection::tryFrom($parameters['value']);
});
```

Hope I managed to clear it up a bit, @bakerkretzmar ?
